File: stack/stackpy/insertAtBegin.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
    

class Stack:
    def __init__(self):
        self.head = None
    
    def push(self, data):
        newNode = Node(data)
        if self.head is None:
            self.head = newNode
            return
        newNode.next = self.head
        self.head = newNode

    def pop(self):
        if self.head is None:
            return
        temp = self.head
        self.head = self.head.next
        del temp
    
    def printStack(self):
        temp = self.head
        while temp:
            print(temp.data, end=" ")
            temp = temp.next
        print()

def main():
    stack = Stack()
    stack.push(10)
    stack.push(20)
    stack.push(30)
    
    print("Stack after inserting elements at the beginning:")
    stack.printStack()
    
    stack.pop()
    
    print("Stack after deleting the top element:")
    stack.printStack()

if __name__ == "__main__":
    main()

File: stack/stackjava/insertAtBegin.java
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class Stack {
    Node top;

    Stack() {
        this.top = null;
    }

    void push(int data) {
        Node newNode = new Node(data);
        newNode.next = top;
        top = newNode;
    }

    int pop() {
        if (top == null) {
            System.out.println("Stack is empty");
            return -1;
        }
        int deleteData = top.data;
        top = top.next;
        return deleteData;
    }
    void display() {
        Node current = top;
        while (current != null) {
            System.out.print(current.data + " ");
            current = current.next;
        }
        System.out.println();
    }
}

public class insertAtBegin {
    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(10);
        stack.push(20);
        stack.push(30);
        System.out.println("Stack after pushing elements:");
        stack.display();

        int popElement = stack.pop();
        System.out.println("Popped element: " + popElement);
        System.out.println("Stack after popping an element:");
        stack.display();
    }
}